pkg/storegateway/bucket_stores_test.go

```diff
@@ -27,6 +28,8 @@ import (
 	"github.com/cortexproject/cortex/pkg/util/flagext"
 )
 
+var _ block.MetadataModifier = &ReplicaLabelRemover{}
```

Comment:
I think this "test" is superfluos. The code wouldn't compile even without this line if `ReplicaLabelRemover` doesn't implement `block.MetadataModifier`. We can remove it.